src/test/java/io/phdata/streamliner/schemadefiner/ScriptCommandTest.java

```diff
@@ -0,0 +1,89 @@
+package io.phdata.streamliner.schemadefiner;
+
+import io.phdata.streamliner.schemadefiner.ConfigBuilder.DiffGenerator;
+import io.phdata.streamliner.schemadefiner.ConfigBuilder.ScriptCommand;
+import io.phdata.streamliner.schemadefiner.model.Configuration;
+import io.phdata.streamliner.schemadefiner.model.ConfigurationDiff;
+import io.phdata.streamliner.schemadefiner.util.StreamlinerUtil;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+
+import static org.junit.Assert.*;
+
+public class ScriptCommandTest {
+  private static String outputPath = "src/test/output/";
+  String templateDirectory = "src/main/resources/templates/snowflake";
+  String typeMapping = "src/main/resources/type-mapping.yml";
+  String outputDir = "src/test/output/pipelineConfig";
+  String configDiff = "src/test/output/confDiff/streamliner-configuration-diff.yml";
+
+  @Rule public ExpectedException expectedEx = ExpectedException.none();
+
+  @Before
+  public void before() {
+    StreamlinerUtil.deleteDirectory(new File(outputPath));
+  }
+
+  // Please comment this after() if want to see the generated output files.
+  @AfterClass
+  public static void after() {
+    StreamlinerUtil.deleteDirectory(new File(outputPath));
+  }
+
+  @Test
+  public void testScriptCommandForSchemaEvolution_Alter_Table_Add_column() throws IOException {
+    Configuration prevConfig =
+        StreamlinerUtil.readYamlFile("src/test/resources/configDiff/addColumn/prevConfig.yml");
+    Configuration currConfig =
+        StreamlinerUtil.readYamlFile("src/test/resources/configDiff/addColumn/currConfig.yml");
+    ConfigurationDiff diff = DiffGenerator.createConfigDiff(prevConfig, currConfig);
+    StreamlinerUtil.writeConfigToYaml(diff, outputPath);
+
+    ScriptCommand.build(null, configDiff, typeMapping, templateDirectory, outputDir);
+    File f = new File("src/test/output/pipelineConfig/Employee");
+    assertTrue(f.exists());
+    assertTrue(f.isDirectory());
+
+    Arrays.stream(f.listFiles())
+        .forEach(
+            file -> {
+              assertTrue(
+                  file.getName().equals("evolve-schema.sql") || file.getName().equals("Makefile"));
+            });
+
+    String actualContent =
+        StreamlinerUtil.readFile("src/test/output/pipelineConfig/Employee/evolve-schema.sql")
+            .trim();
+    String expectedContent = "ALTER TABLE Employee ADD COLUMN Age NUMBER(38, 0);";
```

Comment:
Can we implement "diff" like testing here similar to SQLMorph?

1. Test runs and writes output to src/test/resources
2. Does not fail if there are differences.
3. We'll see and evaluate the differences by git diff.

Reply:
I think if it has not fail then we have to remove the asserts. Only it should call the script command and write the output